includes/class-wc-gateway-ppec-client.php

```diff
@@ -794,13 +794,14 @@ public function do_express_checkout_payment( array $params ) {
 	 * @return array Params for DoExpressCheckoutPayment call
 	 */
 	public function get_do_express_checkout_params( array $args ) {
-		$settings  = wc_gateway_ppec()->settings;
-		$order     = wc_get_order( $args['order_id'] );
+		$settings     = wc_gateway_ppec()->settings;
+		$order        = wc_get_order( $args['order_id'] );
 
-		$old_wc    = version_compare( WC_VERSION, '3.0', '<' );
-		$order_id  = $old_wc ? $order->id : $order->get_id();
-		$details   = $this->_get_details_from_order( $order_id );
-		$order_key = $old_wc ? $order->order_key : $order->get_order_key();
+		$old_wc       = version_compare( WC_VERSION, '3.0', '<' );
+		$order_id     = $old_wc ? $order->id : $order->get_id();
+		$order_number = $old_wc ? (string) apply_filters( 'woocommerce_order_number', $order->get_id(), $order ) : $order->get_order_number();
```

Comment:
We can use `$order_number = $order->get_order_number();`, it's defined on both WC 2.6 and WC 3.x :)